GrandCentralBoard/Widgets/Bonus/Scene/Bubble.swift

```diff
@@ -47,26 +47,28 @@ class Bubble: SKSpriteNode {
     }
     
     func updateWithNewBonus(newBonus: Int) {
+        
         let difference = newBonus - self.bonus
-
+        self.bonus += difference
+        
         // We increase bonus and run animation only if the value of bonus changes for a bigger one.
         guard difference > 0 else { return }
         
-        self.bonus += difference
+        let scaleActionKey = "scaleAction"
+        let shakeActionKey = "shakeAction"
         
-        var scaleBy: CGFloat = 2.3
-        let scaleUpAction = SKAction.scaleBy(scaleBy, duration: 0.5)
-        let scaleDownAction = SKAction.scaleBy(1/scaleBy, duration: 0.1)
+        let isAlreadyScaling = (self.actionForKey(scaleActionKey) != nil)
+        if isAlreadyScaling { return }
```

Comment:
Is it possible to keep the bubble scaling for 15 seconds (the original amount of time) from the moment it received a bonus again?